### DISCUSSION TITLE
Does actions/checkout count against Git LFS bandwidth usage?

### DISCUSSION THREAD

**Discussion by jaydenmilne** [community/community#26775] [Actions]
<p>We are considering migrating our CI infrastructure from a locally hosted Git LFS/Jenkins Pipelines instance to GitHub hosted LFS and GitHub Actions.</p>
<p>We want to know if using <a href="https://github.com/actions/checkout" rel="nofollow noopener">actions/checkout</a> with lfs: true will count against our LFS bandwith quota. It seems like it shouldn’t since it is internal bandwidth on GitHub’s network, and with the pricing for LFS bandwith using it with any Actions pipeline would be cost prohibitive.</p>
<p>There is <a href="https://github.community/t/is-pulling-github-git-lfs-in-github-actions-count-towards-github-git-fls-bandwidth-quota/16618">only one answer</a> from GitHub Actions first launched that has contradictory responses, <a class="mention" href="/chrispat">@chrispat</a> said that it will, but <a class="mention" href="/wildone">@wildone</a> in the comments mentioned that it currently wasn’t charging them.</p>
<p>So:</p>
<ol>
<li>What is the official policy on our organization being billed for “outgoing” LFS bandwitdth that is internal to GitHub’s network?</li>
<li>If it isn’t being billed right now, is that an implementation detail subject to change?</li>
</ol>
<p>It would be nice if the policy on this was clarified in the documentation somewhere (like <a href="https://help.github.com/en/github/setting-up-and-managing-billing-and-payments-on-github/about-billing-for-git-large-file-storage" rel="nofollow noopener">the page on billing for GitHub LFS usage</a> or <a href="https://help.github.com/en/github/setting-up-and-managing-billing-and-payments-on-github/about-billing-for-github-actions" rel="nofollow noopener">billing for GitHub Actions</a>) instead of being buried in the forum.</p>

**Comment by Yanjingzhu** [comment]
<p>Thank you for your question. Based on my test, after I checked out a 142MB LFS file in my workflow run, the Git LFS Data bandwidth is still showing 0. I have contacted github engineering team , when they give me response, I will update here as soon as possible. </p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p><a class="mention" href="/jaydenmilne">@jaydenmilne</a>  I have confirmed that actions/checkout is counted into Git LFS bandwidth usage. </p>
<p>The answer is that every download for every purpose is counted against LFS usage. The SyncAssetUsage job is what computes this data, and it knows nothing about Actions or other special cases.</p>
<p>You could check the LFS usage in your organization settings &gt; Billing page.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/e/e8e335e743915e89708b25db234ef78cafe8a80e.png" data-download-href="/uploads/short-url/xedzjfwk13q1UOUnsss7BbZWL82.png?dl=1" title="lfs bandwidth.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181099855-4c654529-6b41-4ae8-829d-cc75c26a32ad.png" alt="lfs bandwidth.png" data-base62-sha1="xedzjfwk13q1UOUnsss7BbZWL82" width="690" height="400" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/e/e8e335e743915e89708b25db234ef78cafe8a80e_2_690x400.png, https://aws1.discourse-cdn.com/github/original/2X/e/e8e335e743915e89708b25db234ef78cafe8a80e.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/e/e8e335e743915e89708b25db234ef78cafe8a80e.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/e/e8e335e743915e89708b25db234ef78cafe8a80e_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">lfs bandwidth.png</span><span class="informations">1025×595 38.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>What I need to mention is that as the SyncAssetUsage job is a singleton, it runs every so many minutes to process data, and that results in a slight processing delay in Billing page.</p>

**Comment by jaydenmilne** [comment]
<p>Thank you for looking into this for me and the thorough response! It would be nice if this were reflected in the official documentation so people don’t have to hunt through the community forum to find it. Is there someone else I shoud contact to give feedback on the documentation?</p>
<p>And for what it’s worth, it would be very nice if it Actions didn’t count against your LFS bandwidth :wink:</p>

**Comment by Yanjingzhu** [comment]
<p>I have already asked to declare that on official document. </p>
<p>For your requirement:  <em>it would be very nice if it Actions didn’t count against your LFS bandwidth </em></p>
<p>I would encourage you to share your idea in the <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">Feedback form for GitHub Actions</a>. </p>

**Comment by portalerprk** [comment]
<p>It’s inconceivable to me that LFS pulls during GitHub Actions would count towards LFS bandwidth… first… you would assume its within GitHub’s own networks… so why am I paying for them to tranfser files between their own devices?</p>
<p>Second, GitHub Packages that are pulled during GitHub Actions <em>do not</em> count towards their bandwidth… so why the difference in billing policy?</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/0/0655c09036ab33fdb8e601b896850aa035d11a97.png" data-download-href="/uploads/short-url/U2AswlX4fBJFBYz3fRgQ8S7L71.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/60405799/181099857-1e11b1ee-a3cf-4c9b-a9c1-b0ed1075d550.png" alt="image" data-base62-sha1="U2AswlX4fBJFBYz3fRgQ8S7L71" width="690" height="463" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/0/0655c09036ab33fdb8e601b896850aa035d11a97_2_690x463.png, https://aws1.discourse-cdn.com/github/optimized/2X/0/0655c09036ab33fdb8e601b896850aa035d11a97_2_1035x694.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/0/0655c09036ab33fdb8e601b896850aa035d11a97_2_1380x926.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/0/0655c09036ab33fdb8e601b896850aa035d11a97_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">2499×1680 278 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by hoonoh** [comment]
<p>I think LFS bandwidth used within actions should be at least counted with smaller ratio since it should be using internal network bandwidth.</p>
<p>As a workaround to save LFS bandwidth, try checking out without lfs option, cache <code>.git/lfs</code> and run <code>git lfs pull</code> after <code>actions/cache</code> step.</p>

**Comment by naavis** [comment]
<p>I was also pretty surprised that checking out code in Github Actions counts towards LFS bandwidth usage. I managed to use up my free LFS bandwidth quota for a month by building one LFS-heavy site a couple of times with Actions. This really turns me off using Github for my projects that require LFS, and maybe migrate to Gitlab or something else.</p>

**Comment by nschloe** [comment]
<p>I just published an alternative gh-actions checkout action that <em>caches</em> all the LFS data, <a href="https://github.com/nschloe/action-cached-lfs-checkout" class="inline-onebox" rel="noopener nofollow ugc">GitHub - nschloe/action-cached-lfs-checkout: GitHub checkout action with LFS files pulled from cache</a>. Instead of</p>
<pre><code class="lang-yaml">- name: Checkout code
  uses: actions/checkout@v2
  with:
    lfs: true
</code></pre>
<p>simply use</p>
<pre><code class="lang-yaml">- name: Checkout code
  uses: nschloe/action-cached-lfs-checkout@v1
</code></pre>
<p>Hope this helps!</p>

**Comment by f4exb** [comment]
<p>Alas this does not work with submodules.</p>

**Comment by asherritt** [comment]
Hi. Are there any updates to this, or work arounds? I created a workflow for a client who needs to version and process videos. Everything works great but this issues is exhausting their monthly bandwidth so fast, and I'm not sure if adding extra bandwidth is going to be the answer as it may be too cost prohibitive. Is there maybe an unlimited plan? Another way like above to avoid having checkouts count against bandwidth?
